rest/doc_api.go

```diff
@@ -252,6 +252,9 @@ func (h *handler) handlePutDoc() error {
 	if err != nil {
 		return err
 	}
+	if body == nil {
+		return base.HTTPErrorf(http.StatusBadRequest, "Document Body is nil")
```

Comment:
I think this error text might be better as "Document body is empty" - what do you think?